### DISCUSSION TITLE
merge 2 svg, position issue

### DISCUSSION THREAD

**Discussion by Noeilnoeil17** [vvvv/SVG#1051] [Q&A] [answered]
Hello everyone, I don't understand why i can't change position of my second svg
It is always starting on top left, i would like add logo in center of qrcode

            SvgDocument doc = new SvgDocument();
            var group = new SvgGroup();
            doc.Children.Add(group);
            SvgDocument qr = SvgDocument.Open("qr.svg");
            qr.Width = 300;
            qr.Height = 300;
            SvgDocument logo = SvgDocument.Open("logoI.svg");
            logo.Width = 50;
            logo.Height = 50;
            logo.X = 125;
            logo.Y = 125;
            group.Children.Add(qr);
            group.Children.Add(logo);
            doc.Write("importtest.svg");

![importtest](https://user-images.githubusercontent.com/120451044/231896870-70057bc9-a621-4784-804a-ea7c39ff31ad.svg)

if somehone can help me, thanks a lot

Thomas

**Comment by H1Gdev** [comment, marked as answer]
```cs
var doc = new SvgDocument();
var group = new SvgGroup();
doc.Children.Add(group);
var qrO = SvgDocument.Open(@"qr.svg");
var qr = (SvgFragment)qrO.DeepCopy<SvgFragment>(); // Optional this time.
qr.Width = 300;
qr.Height = 300;
var logoO = SvgDocument.Open(@"logoI.svg");
var logo = (SvgFragment)logoO.DeepCopy<SvgFragment>();
logo.Width = 50;
logo.Height = 50;
logo.X = 125;
logo.Y = 125;
group.Children.Add(qr);
group.Children.Add(logo);
doc.Write(@"importtest.svg");
```

`SvgDocument` has characteristics of root \<svg\>.
So `X`, `Y` properties do not work.
By duplicating `SvgFragment` from `SvgDocument`, it can be inner \<svg\>.

**Comment by H1Gdev** [comment]
If cost of copying is high, there is also method of increasing one layer.

```cs
var logoO = SvgDocument.Open(@"logoI.svg");
//var logo = (SvgFragment)logoO.DeepCopy<SvgFragment>();
var logo = new SvgFragment();
logo.Children.Add(logoO);
logo.Width = 50;
logo.Height = 50;
```

**Comment by H1Gdev** [comment]
Third plan.

```cs
var logoO = SvgDocument.Open(@"logoI.svg");
//var logo = (SvgFragment)logoO.DeepCopy<SvgFragment>();
var logo = new SvgFragment();
foreach (var element in logoO.Children)
    logo.Children.Add(element);
logo.Width = 50;
logo.Height = 50;
```

**Comment by Noeilnoeil17** [comment]
Hi H1Gdev,

thanks a lot for your help, your solutions are ok for me. No cost time in my case so first solution is perfect
And thanks for your explanation about svgFragment, community is so helpful
have a nice day

Thomas

**Reply by mrbean-bremen** [reply]
Helpful discussion items shall stay open - reopened.
